src/EnergyPlus/HVACControllers.cc

```diff
@@ -1130,49 +1145,49 @@ namespace HVACControllers {
 			//          earlier on the air loop.
-			if ( ! ControllerProps( ControlNum ).IsSetPointDefinedFlag ) {
-				// NOTE: For TEMPANDHUMRAT control the computed value ControllerProps(ControlNum)%SetPointValue
-				//       depends on:
-				//       - Node(SensedNode)%HumRatMax
-				//       - Node(SensedNode)%Temp
-				//       - Node(SensedNode)%HumRat
-				if ( Node( SensedNode ).HumRatMax > 0 ) {
-					// Setpoint can only be computed when the sensed node temperature is evaluated at the max
-					// actuated value for the dual humidity ratio / temperature strategy.
-					// See routine CalcSimpleController() for the sequence of operations.
-					if ( ControllerProps( ControlNum ).NextActuatedValue == RootFinders( ControlNum ).MaxPoint.X ) {
-						// Calculate the approach temperature (difference between SA dry-bulb temp and SA dew point temp)
-						ApproachTemp = Node( SensedNode ).Temp - PsyTdpFnWPb( Node( SensedNode ).HumRat, OutBaroPress );
-						// Calculate the dew point temperature at the SA humidity ratio setpoint
-						DesiredDewPoint = PsyTdpFnWPb( Node( SensedNode ).HumRatMax, OutBaroPress );
-						// Adjust the calculated dew point temperature by the approach temp
-						HumidityControlTempSetPoint = DesiredDewPoint + ApproachTemp;
-						// NOTE: The next line introduces a potential discontinuity into the residual function
-						//       which could prevent the root finder from finding the root it if were done at each
-						//       controller iteration. For this reason we perform the setpoint calculation only
-						//       once when the air loop has been evaluated with the max actuated value.
-						//       See routine CalcSimpleController() for the sequence of operations.
-						ControllerProps( ControlNum ).SetPointValue = min( Node( SensedNode ).TempSetPoint, HumidityControlTempSetPoint ); // Pure temperature setpoint | Temperature setpoint to achieve the humidity ratio setpoint
-						// Overwrite the "pure" temperature setpoint with the actual setpoint that takes into
-						// account the humidity ratio setpoint.
-						// NOTE: Check that this does not create side-effects somewhere else in the code.
-						Node( SensedNode ).TempSetPoint = ControllerProps( ControlNum ).SetPointValue;
-						// Finally indicate thate the setpoint has been computed
-						ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
-					}
-				} else {
-					// Pure temperature setpoint control strategy
-					ControllerProps( ControlNum ).SetPointValue = Node( SensedNode ).TempSetPoint;
-					// Finally indicate thate the setpoint has been computed
-					ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
-				}
+			// NOTE: For TEMPANDHUMRAT control the computed value ControllerProps(ControlNum)%SetPointValue
+			//       depends on:
+			//       - Node(SensedNode)%HumRatMax
+			//       - Node(SensedNode)%Temp
+			//       - Node(SensedNode)%HumRat
+			if ( Node( SensedNode ).HumRatMax > 0 ) {
```

Comment:
I made that change and tested the defect file. Results look promising. If you do run the defect file, lower the HumRat SP from 0.0075 to at least 0.006 or you won't see any humidity control. I got rid of all variables except for 

  Output:Variable,VAV 1 Cooling Coil Outlet,System Node Temperature,hourly;
  Output:Variable,VAV 1 Cooling Coil Outlet,System Node Setpoint Temperature,hourly;
  Output:Variable,VAV 1 Cooling Coil Outlet,System Node Setpoint Maximum Humidity Ratio,hourly;
  Output:Variable,VAV 1 Cooling Coil Outlet,System Node Humidity Ratio,hourly;

  Output:Variable,VAV 2 Cooling Coil Outlet,System Node Temperature,hourly;
  Output:Variable,VAV 2 Cooling Coil Outlet,System Node Setpoint Temperature,hourly;
  Output:Variable,VAV 2 Cooling Coil Outlet,System Node Setpoint Maximum Humidity Ratio,hourly;
  Output:Variable,VAV 2 Cooling Coil Outlet,System Node Humidity Ratio,hourly;

  Output:Variable,VAV 3 Cooling Coil Outlet,System Node Temperature,hourly;
  Output:Variable,VAV 3 Cooling Coil Outlet,System Node Setpoint Temperature,hourly;
  Output:Variable,VAV 3 Cooling Coil Outlet,System Node Setpoint Maximum Humidity Ratio,hourly;
  Output:Variable,VAV 3 Cooling Coil Outlet,System Node Humidity Ratio,hourly;